src/Klein/Response.php

```diff
@@ -92,9 +95,13 @@ public function file($path, $filename = null, $mimetype = null)
         $this->header('Content-length', filesize($path));
         $this->header('Content-Disposition', 'attachment; filename="'.$filename.'"');
 
-        $this->send();
+        $bytes_read = readfile($path);
 
-        readfile($path);
+        if (false === $bytes_read) {
+            throw new RuntimeException('The file could not be read');
+        }
+
+        $this->send();
```

Comment:
Hmm, now that I think about this, this won't work if output buffering is disabled or if the buffer has already closed, sent, or potentially even cleared.

If output buffering is disabled or the buffer is closed, then the `readfile()` call will directly output the file's contents to the HTTP response **before HTTP headers are sent**. That will form an incorrect HTTP response. Now, PHP's default runtime _will_ prevent that, but in doing so it'll still return an error when we try to send the headers that are modeled in the current response object.

So this isn't a solution that we can depend on, unfortunately.